pkg/utils/test/integration.go

```diff
@@ -23,17 +31,21 @@ const (
 	// ControlPlaneStartTimeout is the time to wait for control plane startup
 	// in kubebuilder integration tests.
 	// It is set at a relatively high value due to low resources in continuous integration.
-	ControlPlaneStartTimeout = 2 * time.Minute
-	BootstrapTestEnvRetries  = 3
+	ControlPlaneStartTimeout = 1 * time.Minute
+	BootstrapTestEnvRetries  = 1
+
+	CRDsRelativePath = "../../../config/crds-patches/all-in-one.yaml"
 )
 
 var Config *rest.Config
 
 // RunWithK8s starts a local Kubernetes server and runs tests in m.
-func RunWithK8s(m *testing.M, crdPath string) {
+func RunWithK8s(m *testing.M) {
+	// add CRDs scheme to the client
 	_ = controllerscheme.SetupScheme()
+
 	t := &envtest.Environment{
-		CRDDirectoryPaths:        []string{crdPath},
+		CRDs:                     parseCRDs(),
```

Comment:
If we keep on reading CRDs from `crds/` we have a problem when applying resources with a `nil` list of containers.
So it makes sense to use the patched merged version of the CRDs instead. But since all CRDs are now in a single file, we need to parse them manually.

Reply:
>But since all CRDs are now in a single file, we need to parse them manually.

Does envtest require 1 file = 1 resource? It looks like it should handle our all-in-one crd so I would have thought we could continue using `CRDDirectoryPaths`
https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/envtest/crd.go#L201

Reply:
You're right, it works! I must have misread the code. Happy to remove that whole new block of code, thanks for looking at it closely.